sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -31,6 +31,8 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
 
   private var _analyzed: Boolean = false
 
+  private var _incremental: Boolean = false
```

Comment:
Adding it here will break `sameResult`, `equals` and other methods. Could you add a new parameter to the constructor of `LogicalRelation` and `LogicalRDD` instead?